routes/auth.routes.js

```diff
@@ -0,0 +1,96 @@
+const mongoose = require('mongoose');
+const {Router} = require ('express')
+const router = new Router()
+const bcryptjs = require('bcryptjs')
+const saltRounds =10
+const User = require ('../models/User.model')
+
+// GET route ==> to display the signup form to users
+router.get ('/signup', (req,res) => res.render('signup.hbs'))
+
+// POST route ==> to process form data
+router.post('/signup', (req, res, next) => { 
+const {username, password} =req.body;
+
+// make sure users fill all mandatory fields:
+if (!username || !password) {
+    res.render('signup', { errorMessage: 'All fields are mandatory. Please provide your username and password.' });
+    return;
+  }
+
+  const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
+  if (!regex.test(password)) {
+    res
+      .status(500)
+      .render('signup', { errorMessage: 'Password needs to have at least 6 chars and must contain at least one number, one lowercase and one uppercase letter.' });
+    return;
+  }
+
+  bcryptjs
+  .genSalt(saltRounds)
+  .then(salt => bcryptjs.hash(password, salt))
+  .then (hashedPassword => {
+  console.log(`Password hash: ${hashedPassword}`)
+  return User.create ({
+      username,
+      password: hashedPassword
+  })
+})
+.then(userFromDB =>{
+    console.log('Newly created user is: ', userFromDB);
+    res.redirect('/userProfile');  
```

Comment:
if you don't set a session, then you should not redirect to the profile; it would be better to redirect to the login page